lib/DXIL/DxilModule.cpp

```diff
@@ -2081,4 +2111,157 @@ bool DxilModule::IsPrecise(const Instruction *inst) const {
     return false;
 }
 
+bool DxilModule::ShaderCompatInfo::Merge(ShaderCompatInfo &other) {
+  bool changed = DXIL::UpdateToMaxOfVersions(minMajor, minMinor, other.minMajor,
+                                             other.minMinor);
+  if ((mask & other.mask) != mask) {
+    mask &= other.mask;
+    changed = true;
+  }
+  uint64_t rawBefore = shaderFlags.GetShaderFlagsRaw();
+  shaderFlags.CombineShaderFlags(other.shaderFlags);
+  if (rawBefore != shaderFlags.GetShaderFlagsRaw())
+    changed = true;
+  return changed;
+}
+
+void DxilModule::ClearShaderCompatInfo() { m_FuncToShaderCompat.clear(); }
```

Comment:
Yeah, it was intended to be used after changes are made to the functions in the module that might require recomputation, such as after a PIX pass or something like that, but I'll call it at the beginning of `CollectShaderFlagsForModule`, since that will also guarantee it's fresh when computing the overall module flags.